### ISSUE TITLE
vbnet:S4144 :  throws an exception of type 'System.NullReferenceException'

### ISSUE THREAD

**Issue by W3bCorp** [SonarSource/sonar-dotnet#2594]
### Description

When i build with TFS2018/VS2017/MsBuild 15 with SonarQube 7.9.1 and Sonar-scanner 3.3.0.1492, i got the error : 

2019-08-20T10:03:09.9054057Z ##[error]vbc(0,0): Error AD0001: Analyzer 'SonarAnalyzer.Rules.VisualBasic.MethodsShouldNotHaveIdenticalImplementations' threw an exception of type 'System.NullReferenceException' with message 'La référence d'objet n'est pas définie à une instance d'un objet.'.
2019-08-20T10:03:09.9054717Z ##[debug]Processed: ##vso[task.logissue type=Error;sourcepath=vbc;linenumber=0;columnnumber=0;code=AD0001;]Analyzer 'SonarAnalyzer.Rules.VisualBasic.MethodsShouldNotHaveIdenticalImplementations' threw an exception of type 'System.NullReferenceException' with message 'La référence d'objet n'est pas définie à une instance d'un objet.'.
2019-08-20T10:03:09.9060360Z vbc : error AD0001: Analyzer 'SonarAnalyzer.Rules.VisualBasic.MethodsShouldNotHaveIdenticalImplementations' threw an exception of type 'System.NullReferenceException' with message 'La référence d'objet n'est pas définie à une instance d'un objet.'. [G:\Builds\agents\day01\_work\1\s\IsiFrameWork\IsiBusinessLogic\IsiBusinessLogic.vbproj]
2019-08-20T10:03:09.9268948Z ##[debug]Processed: ##vso[task.logdetail id=9765be38-5431-4b86-b27f-4e154cb645ee;parentid=a570503f-71e3-4c21-adc8-32d46823bbd8;type=Build;result=Failed;finishtime=2019-08-20T10:03:09.9265529Z;progress=100;state=Completed;parentid=a570503f-71e3-4c21-adc8-32d46823bbd8;name=;]

### Repro steps

Standard build, with standard "Sonar way" quality profile. 
The repro is difficult because +100k of lines of code
When i disabled the rule "vbnet:S4144", all is ok.

I try to diagnostic with "sonar.verbose=true" and msbuild "-v:diag" args but i have no more informations. Visibly, the error occurs randomly and not always in the same file

### Related information

* SonarQube : 7.9.1
* Sonar-scanner : 3.3.0.1492
* Visual Studio : 2017
* MsBuild : 15
* TFS : 2018

**Comment by andrei-epure-sonarsource** [comment]
thanks @W3bCorp . It's hard w/o a reproducer, but I'm marking as a bug to pick up in one of our future sprints. If we'll be able to find the issue by reading the code or finding a repro, we'll fix it.

**Comment by andrei-epure-sonarsource** [comment]
Possibly related from `staxrip` [cirrus ci job](https://api.cirrus-ci.com/v1/task/5112645952208896/logs/build.log)
```
vbc : error AD0001: Analyzer 'SonarAnalyzer.Rules.VisualBasic.ThreadResumeOrSuspendShouldNotBeCalled' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [C:\Project\StaxRip.vbproj]
vbc : error AD0001: Analyzer 'SonarAnalyzer.Rules.VisualBasic.DangerousGetHandleShouldNotBeCalled' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [C:\Project\StaxRip.vbproj]
vbc : error AD0001: Analyzer 'SonarAnalyzer.Rules.VisualBasic.OptionalParameterNotPassedToBaseCall' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [C:\Project\StaxRip.vbproj]
    7379 Warning(s)
    3 Error(s)

```

**Comment by costin-zaharia-sonarsource** [comment]
Adding the stack trace:
```
Warning	AD0001	Analyzer 'SonarAnalyzer.Rules.VisualBasic.OptionalParameterNotPassedToBaseCall' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.	StaxRip		1		Active	Analyzer 'SonarAnalyzer.Rules.VisualBasic.OptionalParameterNotPassedToBaseCall' threw the following exception:
'Exception occurred with following context:
Compilation: StaxRip
SyntaxTree: C:\src\staxrip\General\Misc.vb
SyntaxNode: (0) [InvocationExpressionSyntax]@[40598..40601) (1029,33)-(1029,36)

System.NullReferenceException: Object reference not set to an instance of an object.
   at SonarAnalyzer.Helpers.VisualBasic.VisualBasicSyntaxHelper.IsOn(ExpressionSyntax expression, SyntaxKind onKind)
   at SonarAnalyzer.Helpers.VisualBasic.VisualBasicSyntaxHelper.IsOn(ExpressionSyntax expression, SyntaxKind onKind)
   at SonarAnalyzer.Helpers.VisualBasic.VisualBasicSyntaxHelper.IsOnBase(ExpressionSyntax expression)
   at SonarAnalyzer.Rules.VisualBasic.OptionalParameterNotPassedToBaseCall.<Initialize>b__6_0(SyntaxNodeAnalysisContext c)
   at SonarAnalyzer.Helpers.DiagnosticAnalyzerContextHelper.<>c__DisplayClass0_0`1.<RegisterSyntaxNodeActionInNonGenerated>b__0(SyntaxNodeAnalysisContext c)
   at SonarAnalyzer.Helpers.SonarAnalysisContext.<>c__DisplayClass30_0`1.<RegisterContextAction>b__0(TContext c)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__53`1.<ExecuteSyntaxNodeAction>b__53_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
-----
'.
```

**Comment by costin-zaharia-sonarsource** [comment]
Fixed by #2735
